src/term/net_to_term.rs

```diff
@@ -65,6 +80,8 @@ pub struct Reader<'a> {
   seen_fans: Scope,
   seen: HashSet<Port>,
   errors: Vec<ReadbackError>,
+  vars: HashSet<Name>,
```

Comment:
I think it would be better to move the gathering of unscopeds to a separate funcion to not increase the complexity of the Reader, which is already quite complex.

We could in addition make the reader always return unscoped lambdas only, to make it's result correct even before checking the scope of variables